### PR TITLE
Mandates referencing issue in PR

### PR DESCRIPTION
**Pull request by onlinejudge95**
Resolves:  #1716 
Closes: #1716 
Fixes: #1716

### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.

### CODE REVIEW

**Comment by onlinejudge95** [comment]
Fixes #1716

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -17,3 +17,4 @@
 * [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
 * [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
 * [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
+* [ ] I am referencing the issue with this PR for auto cleanup. For eg. `fixes: #{$ISSUE_NO}`
```
Comment (cclauss) [inline .github/pull_request_template.md]: If this issues resolves an open issue then the commit message contains `fixes: #{$ISSUE_NO}` for auto cleanup.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -25,6 +25,8 @@ We appreciate any contribution, from fixing a grammar mistake in a comment to im
 
 Your contribution will be tested by our [automated testing on Travis CI](https://travis-ci.org/TheAlgorithms/Python/pull_requests) to save time and mental energy.  After you have submitted your pull request, you should see the Travis tests start to run at the bottom of your submission page.  If those tests fail, then click on the ___details___ button try to read through the Travis output to understand the failure.  If you do not understand, please leave a comment on your submission page and a community member will try to help.
 
+It would help us a lot in cleanup if you are mentioning your Issue no. in your Pull Request. For eg. Adding this text(`fixes: #{$ISSUE_NO}`) as a comment in your PR
```
Comment (cclauss) [inline CONTRIBUTING.md]: Please help us keep our issue list small by adding `fixes: #{$ISSUE_NO}` to the commit message of pull requests that resolve open issues.  GitHub will use this tag to auto close the issue when the PR is merged.

Comment (poyea) [reply CONTRIBUTING.md]: You're right. I think that's fine if we just mention adding it to the PR itself.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -17,3 +17,4 @@
 * [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
 * [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
 * [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
+* [ ] I am referencing an issue in this PR or in the commit message for auto cleanup. For example, `fixes: #1234` or `closes: #567`.
```
Comment (cclauss) [inline .github/pull_request_template.md]: __If a PR resolves an issue then__ it should contain a link to that issue.

I do not want to mandate that all PRs _need_ an issue first.  I find that to be an inefficient approach.

Comment (cclauss) [reply .github/pull_request_template.md]: See the commit message above... `Resolves:` also works.

**Pull request merged**